### PR TITLE
Added bubblesort implementation

### PR DESCRIPTION
**Pull request by anubhavsrivastava**
Reference #44

### CODE REVIEW

File: c/sort/bubblesort_optimised.c [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    int i;
+    for (i=0; i < size; i++)
+        printf("%d ", arr[i]);
+    printf("\n");
+}
+
+// Driver program to test above functions
+int main()
+{
+    int arr[] = {64, 34, 25, 12, 22, 11, 90};
```
Comment (thecodershub) [inline c/sort/bubblesort_optimised.c]: Please add some more tests like empty arrays, negative elements, etc.

**Review by thecodershub — COMMENTED** [review]
Kindly add assert statements too, as has been added for other languages like Python.

**Comment by anubhavsrivastava** [comment]
assert is not provided by default. There will be huge code for asserting. I would suggest to keep it straight in case of c/cpp

**Comment by thecodershub** [comment]
Doesn't the header `assert.h` provide a macro for this? We are adding asserts everywhere because in future, we will add automated testing to this repository.

**Comment by anubhavsrivastava** [comment]
it does. That's what is used for assertion. But if you want it to be testable, main purpose for algo should be modular and export a function for assertion. You may have same data set for all languages for automated testing.

**Comment by thecodershub** [comment]
We don't just intend to include sorting. We might have more complex algorithms in the future. Same dataset won't work.

**Comment by anubhavsrivastava** [comment]
same dataset for all language for each individual algorithm!

**Comment by thecodershub** [comment]
So you mean to say that for every algorithm-language, you expose a function that we give to a tester function or we tell every contributor to have common function names for easy testing? How's that better over assert? And is assert really that heavy to be included?

**Comment by anubhavsrivastava** [comment]
that is a way you see it. common function names aren't required though. You can have testing setup done before for each language. how is it better? seriously?  assert library may sound light weight but it is not. And for every other program do you thing each contribution would be asserted in same way?

**Comment by thecodershub** [comment]
We are here following a pattern to add tests with simple assertions in same code itself. Your suggestion is good and we might think on it in future. For now please follow common pattern of assertions.
